### PR TITLE
Fix issue with Installed Apps trying to load invalid action links

### LINKED ISSUE #6380: Apps with an invalid `docs_view_name` cause the entire installed apps page to fail
### Environment
* Nautobot version (Docker tag too if applicable): v2.3.7
* Python version:
* Database platform, version:
* Middleware(s):

### Steps to Reproduce
1. change the `docs_view_name` in `NautobotAppConfig` for any app to a name that doesn't exist as a view name in Django
2. Go to `/apps/installed-apps/`

### Expected Behavior
Docs link is skipped, the same behavior is `docs_view_name` is not provided

### Observed Behavior
500 error:
image

### PR DESCRIPTION
**Pull request by smk4664**
# Closes #6380 
# What's Changed
Catch NoReverseMatch exception.

# Screenshots

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [x] Example App Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by timizuoebideri1 — APPROVED** [review]

File: nautobot/extras/plugins/views.py [hunk 1]
```diff
@@ -30,6 +30,18 @@ def get(self, request):
...
+                except NoReverseMatch:
+                    app.home_view_name = None
+                try:
+                    reverse(app.config_view_name)
+                except NoReverseMatch:
+                    app.config_view_name = None
+                try:
+                    reverse(app.docs_view_name)
+                except NoReverseMatch:
+                    app.docs_view_name = None
```
Comment (glennmatthews) [inline nautobot/extras/plugins/views.py:44]: I'm not 100% comfortable with this actually modifying the `app` data in the error case. Might be safer/more correct to set `home_view_name`, etc. as local variables?

Comment (smk4664) [reply nautobot/extras/plugins/views.py:44]: Sure!

**Review by glennmatthews — APPROVED** [review]

**Comment by glennmatthews** [comment]
Any chance a similar fix is needed for `/api/apps/installed-apps/`?

**Comment by smk4664** [comment]
> Any chance a similar fix is needed for `/api/apps/installed-apps/`?

It's already in the API view.  

```
    @staticmethod
    def _get_app_data(app_config):
        try:
            home_url = reverse(app_config.home_view_name)
        except NoReverseMatch:
            home_url = None
        try:
            config_url = reverse(app_config.config_view_name)
        except NoReverseMatch:
            config_url = None
        try:
            docs_url = reverse(app_config.docs_view_name)
        except NoReverseMatch:
            docs_url = None
```

**Pull request merged**